test/Microsoft.AspNet.Mvc.FunctionalTests/ModelBindingTest.cs

```diff
@@ -1692,5 +1693,215 @@ public class ModelBindingTest
             var fileContent = await response.Content.ReadAsStringAsync();
             Assert.Equal(expectedContent, fileContent);
         }
+
+        [Fact]
+        public async Task BindModelAsync_WithCollection()
+        {
+            // Arrange
+            var server = TestServer.Create(_services, _app);
+            var client = server.CreateClient();
+            var content = new Dictionary<string, string>
+            {
+                { "AddressLines[0].Line", "Street Address 0" },
+                { "AddressLines[1].Line", "Street Address 1" },
+                { "ZipCode", "98052" },
+            };
+            var url = "http://localhost/BindModel/CollectionType";
+            var formData = new FormUrlEncodedContent(content);
+
+            // Act
+            var response = await client.PutAsync(url, formData);
+
+            // Assert
+            var address = JsonConvert.DeserializeObject<Address2>(
```

Comment:
We should verify the status code before reading this. Can we move this to a separate method that does : 

```C#
public async Task<TVal> ReadValue<TVal>(HttpResponseMessage response)
{
   response.EnsureSuccessfulStatusCode();
   return JsonConvert.DeserializeObject<TVal>(await response.Content.ReadAsStringAsync());
}
```